src/generators/typescript/TypeScriptRenderer.ts

```diff
@@ -96,16 +100,32 @@ ${lines.map(line => ` * ${line}`).join('\n')}
       content.push(rendererProperty);
     }
 
+    if (this.model.additionalProperties !== undefined) {
+      const propertyName = getUniquePropertyName(this.model, DefaultPropertyNames.additionalProperties);
+      const additionalProperty = await this.runPropertyPreset(propertyName, this.model.additionalProperties, PropertyType.additionalProperty);
+      content.push(additionalProperty);
+    }
+
     return this.renderBlock(content);
   }
 
-  renderProperty(propertyName: string, property: CommonModel): string {
+  renderProperty(propertyName: string, property: CommonModel, type: PropertyType = PropertyType.property): string {
     const name = FormatHelpers.toCamelCase(propertyName);
-    const signature = this.renderTypeSignature(property, { isRequired: this.model.isRequired(propertyName) });
-    return `${name}${signature};`;
+    let signature: string;
+    switch (type) {
+    case PropertyType.property:
+      signature = this.renderTypeSignature(property, { isRequired: this.model.isRequired(propertyName) });
+      return `${name}${signature};`;
+      break;
+    case PropertyType.additionalProperty:
+      signature = this.renderType(property);
+      return `${name}?: Map<String, ${signature}>;`;
+    default:
+      return '';
+    }
```

Comment:
```suggestion
    switch (type) {
    case PropertyType.property:
      const signature = this.renderTypeSignature(property, { isRequired: this.model.isRequired(propertyName) });
      return `${name}${signature};`;
    case PropertyType.additionalProperty:
      const signature = this.renderType(property);
      return `${name}?: Map<String, ${signature}>;`;
    default:
      return '';
    }
```

Reply:
then we have to disable `no-case-declarations` which causes `Unexpected lexical declaration in case block`

Reply:
So we need this `let signature: string;` outside the switch?